### PR TITLE
L1 descriptor fixes (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Taken from #6673.

Will run tests soon.

### CODE REVIEW

**Comment by korydraughn** [comment]
All unit tests passed.

Running core tests ...

**Review by alanking — COMMENTED** [review]
Seems good

How is this interacting with #6673? Should we include the changes from #6793 here, too?

**Comment by korydraughn** [comment]
I'm going to handle the deprecation in a separate PR since I've already tested this.

I expect there will be conflicts with the ASan PR, but those will be simple to resolve.

**Review by alanking — DISMISSED** [review]
In that case, looks good to me. Shall we await other requested reviewers?

**Comment by korydraughn** [comment]
Let's give it an hour. If no one shows up, I'll add the pounds.

File: server/core/src/objDesc.cpp [hunk 1]
```diff
@@ -28,10 +29,47 @@
...
+    _l1d.dataObjInfo = nullptr;
+    _l1d.otherDataObjInfo = nullptr;
+    _l1d.replDataObjInfo = nullptr;
+    _l1d.remoteZoneHost = nullptr;
+
+    _l1d.pluginData.clear();
+    _l1d.replica_token.clear();
+
+    // NOLINTBEGIN(cppcoreguidelines-pro-bounds-array-to-pointer-decay)
+    std::memset(_l1d.chksum, 0, sizeof(l1desc::chksum));
```
Comment (MartinFlores751) [inline server/core/src/objDesc.cpp:68]: Might be worth considering `std::fill` vs `std::memset`.

**Review by MartinFlores751 — DISMISSED** [review]
Everything looks great! Just one small thing to think about is all I have to comment.

File: server/core/src/objDesc.cpp (hunk 1 above)
Comment (korydraughn) [reply server/core/src/objDesc.cpp:68]: I could use `std::fill`, but `std::memset` seems perfectly fine for this case given `l1desc::chksum` is a plain ole C array.

Is there an argument against using `std::memset`?

Comment (korydraughn) [reply server/core/src/objDesc.cpp:68]: After some discussion, I'm going to move forward with `std::memset`.

There are online articles that hint at the possibility that `std::fill` can produce the same assembly as `std::memset`, but not always.

We should investigate in which cases it is better to use one function over the other.

**Comment by korydraughn** [comment]
#'d

**Review by alanking — APPROVED** [review]

**Pull request merged**